### PR TITLE
feat: add batch search projects multiselect

### PR DESCRIPTION
**Pull request by caro3801**
Related to ICIJ/datashare#992

Add [vue-multiselect](https://vue-multiselect.js.org/#sub-getting-started) with same [style](https://github.com/ICIJ/prophecies/blob/main/prophecies/apps/frontend/src/utils/_multiselect.scss) as done on ICIJ/prophecies => proposal for future steps could be to add it to murmur.

Screenshot from 2023-01-19 12-00-19

Screenshot from 2023-01-19 12-00-45

### CODE REVIEW

File: src/components/BatchSearchForm.vue [hunk 1]
```diff
@@ -4,21 +4,25 @@
...
+            class="batch-search-form__fileLabel mb-0"
+            label-size="sm"
+          >
             <template slot="description">
               <div v-html="$t('batchSearch.fileDescription')"></div>
             </template>
             <b-form-file
               v-model="csvFile"
               :placeholder="$t('batchSearch.filePlaceholder')"
+              :state="Boolean(csvFile)"
```
Comment (pirhoo) [inline src/components/BatchSearchForm.vue:23]: I know it was already there but this state should be a computed property. I would have preferred the shorthand `!!` too.

Comment (pirhoo) [inline src/components/BatchSearchForm.vue:23]: I know it was already there but this state should be a computed property. I would have preferred the shorthand `!!` too.

File: src/components/BatchSearchForm.vue [hunk 2]
```diff
@@ -32,28 +36,28 @@
...
+            :label="`${$t('batchSearch.projects')} *`"
+            class="batch-search-form__projects"
             label-size="sm"
+          >
+            <multiselect
+              v-model="projects"
+              :allow-empty="false"
+              :multiple="true"
+              :options="availableProjects"
+              :searchable="true"
```
Comment (pirhoo) [inline src/components/BatchSearchForm.vue:53]: The props `:multiple="true"` and `:searchable="true"` can be set without `="true"`. Also `searchable` is default to true so we can remove it.

File: src/components/BatchSearchForm.vue [hunk 3]
```diff
@@ -342,12 +358,23 @@ export default {
...
+          each(aggTypes, (aggType) => {
+            const extensions = has(types, aggType) ? types[aggType].extensions : []
+            const label = has(types, aggType) ? types[aggType].label : aggType
+            this.allFileTypes.push({
+              extensions,
+              label,
+              mime: aggType
+            })
+          })
+        } catch (e) {
```
Comment (pirhoo) [inline src/components/BatchSearchForm.vue:372]: Good call catching this error

File: src/components/BatchSearchForm.vue [hunk 4]
```diff
@@ -373,6 +400,7 @@ export default {
     },
     async onSubmit() {
       try {
+        if (!this.projects?.length) return
```
Comment (pirhoo) [inline src/components/BatchSearchForm.vue:403]: Is this necessary? If we have the `allow-blank=false` on the multiselect, couldn't we use the form validation to disable the submit button?

**Review by pirhoo — APPROVED** [review]
Excellent addition, very well done! Thanks @caro3801!

**Review by mvanzalu — APPROVED** [review]

File: src/utils/_multiselect.scss [hunk 5]
```diff
@@ -0,0 +1,115 @@
+@import url('~node_modules/vue-multiselect/dist/vue-multiselect.min.css');
```
[the remark this answers is not in the document]
Comment (pirhoo) [reply src/utils/_multiselect.scss:1]: It is shipped with vue-multiselect yes

File: src/components/BatchSearchForm.vue (hunk 1 above)
Comment (caro3801) [reply src/components/BatchSearchForm.vue]: Me too; it's changed

File: src/components/BatchSearchForm.vue (hunk 2 above)
Comment (caro3801) [reply src/components/BatchSearchForm.vue]: :broom:

File: src/utils/_multiselect.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (caro3801) [reply src/utils/_multiselect.scss:1]: The `vue-multiselect.min.css`  is the original css file from the lib that has been minified (automatically). 
In order to adapt the component style to our project, we added `src/utils/_multiselect.scss` which import the orignal css in order to override its properties.

[the remark this answers is not in the document]
Comment (mvanzalu) [reply src/utils/_multiselect.scss:1]: Got it ! THanks

**Pull request merged**